src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -619,6 +619,10 @@ public int getLevel()
      */
     public void setLevel(final int lvl)
     {
+        if(lvl == 0)
+        {
+            this.experience = 0;
```

Comment:
Can we not just make a second method named `resetExperience` that sets level and experience to 0?